From: ben@algroup.co.uk (Ben Laurie)
Subject: initgroups() stand in did not initialise or clean up
Affects: util.c
Changelog: The replacement for initgroups() did not call {set,end}grent(). This
	had two implications: if anything else used getgrent(), then
	initgroups() would fail, and it was consuming a file descriptor.

*** util.c.old	Mon Dec 11 17:09:45 1995
--- util.c	Mon Dec 11 17:10:24 1995
***************
*** 709,714 ****
--- 709,716 ----
    struct group *g;
    int index = 0;
  
+   setgrent();
+ 
    groups[index++] = basegid;
  
    while (index < NGROUPS_MAX && ((g = getgrent()) != NULL))
***************
*** 720,725 ****
--- 722,729 ----
          if (!strcmp(*names, name))
            groups[index++] = g->gr_gid;
      }
+ 
+   endgrent();
  
    return setgroups(index, groups);
  #endif /* def QNX */

